django/db/backends/mysql/schema.py

```diff
@@ -65,6 +77,14 @@ def _field_should_be_indexed(self, model, field):
             return False
         return create_index
 
+    def _create_index_sql(self, model, fields, **kwargs):
+        statement = super()._create_index_sql(model, fields, **kwargs)
+        if len(fields) == 1 and fields[0].max_length is not None and fields[0].max_length > 191:
+            # Limit single-column index to 191 first chars
+            # (worst case: InnoDB engine and utf8mb4 charset)
+            statement.parts['columns'] = ColumnWithSize(statement.parts['columns'], 191)
```

Comment:
Not a huge deal, but it might be nice to have this user-configurable, defaulting to 191, because some people may increase the maximum length for their database.